<a href="https://colab.research.google.com/github/maycgr/Projeto-Analise-Covid/blob/main/Projeto_AED_MAYARA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img src="https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/main/media/logo/newebac_logo_black_half.png" alt="ebac-logo">

---

# **Módulo** | Análise de Dados: COVID-19 Dashboard
Caderno de **Exercícios**<br>
Professor [André Perez](https://www.linkedin.com/in/andremarcosperez/)

---

# **Tópicos**

<ol type="1">
  <li>Introdução;</li>
  <li>Análise Exploratória de Dados;</li>
  <li>Visualização Interativa de Dados;</li>
  <li>Storytelling.</li>
</ol>


---

# **Exercícios**

Este *notebook* deve servir como um guia para **você continuar** a construção da sua própria análise exploratória de dados interativa. Fique a vontate para copiar os códigos da aula mas busque explorar os dados ao máximo. Por fim, publique seu *notebook* no [Kaggle](https://www.kaggle.com/) e seu *dashboard* [Google Data Studio](https://datastudio.google.com/).

---

# **COVID Dashboard**

 - **Dashboard**:
  - Google Data Studio ([link](https://lookerstudio.google.com/reporting/931ee815-caed-4cd0-bc1f-474081ba287f)).
 - **Processamento**:
  - Kaggle Notebook ([link](https://www.kaggle.com/code/mayaragraciano/projeto-analise-de-dados-covid)).
 - **Fontes**:
  - Casos pela universidade John Hopkins ([link](https://github.com/CSSEGISandData/COVID-19/tree/master/csse_covid_19_data/csse_covid_19_daily_reports));
  - Vacinação pela universidade de Oxford ([link](https://covid.ourworldindata.org/data/owid-covid-data.csv)).

## 1\. Contexto

**COVID-19: Uma Introdução**

A COVID-19, causada pelo coronavírus SARS-CoV-2, é uma doença respiratória que se espalhou globalmente a partir de 2019, resultando em uma pandemia. Transmitida principalmente por gotículas respiratórias, seus sintomas variam de leves a graves, incluindo febre, tosse seca e dificuldade para respirar. A pandemia impactou profundamente a saúde pública, levando a medidas como quarentenas e campanhas de vacinação. A análise de dados tem sido fundamental para monitorar a pandemia e orientar políticas públicas, ajudando na compreensão e controle da doença.

Este projeto busca construir um dashboard de dados para exploração e visualização interativa de dados sobre o avanço de casos e da vacinação do Brasil. O processamento de dados está neste ([link](https://www.kaggle.com/code/mayaragraciano/projeto-analise-de-dados-covid)) e o dashboard, neste ([link](https://lookerstudio.google.com/reporting/931ee815-caed-4cd0-bc1f-474081ba287f)).

**DADOS**

Os dados sobre **casos da COVID-19** são compilados pelo centro de ciência de sistemas e engenharia da universidade americana **John Hopkins** ([link](https://www.jhu.edu)). Os dados são atualizados diariamente deste janeiro de 2020 com uma granularidade temporal de dias e geográfica de regiões de países (estados, condados, etc.). O website do projeto pode ser acessado neste [link](https://systems.jhu.edu/research/public-health/ncov/) enquanto os dados, neste [link](https://github.com/CSSEGISandData/COVID-19/tree/master/csse_covid_19_data/csse_covid_19_daily_reports). Abaixo estão descritos os dados derivados do seu processamento.

- **date**: data de referência;
 - **state**: estado;
 - **country**: país;
 - **population**: população estimada;
 - **confirmed**: número acumulado de infectados;
 - **confirmed_1d**: número diário de infectados;
 - **confirmed_moving_avg_7d**: média móvel de 7 dias do número diário de infectados;
 - **confirmed_moving_avg_7d_rate_14d**: média móvel de 7 dias dividido pela média móvel de 7 dias de 14 dias atrás;
 - **deaths**: número acumulado de mortos;
 - **deaths_1d**: número diário de mortos;
 - **deaths_moving_avg_7d**: média móvel de 7 dias do número diário de mortos;
 - **deaths_moving_avg_7d**: média móvel de 7 dias dividido pela média móvel de 7 dias de 14 dias atrás;
 - **month**: mês de referência;
 - **year**: ano de referência.

Os dados sobre **vacinação da COVID-19** são compilados pelo projeto Nosso Mundo em Dados (*Our World in Data* ou OWID) da universidade britânica de **Oxford** ([link](https://www.ox.ac.uk)). Os dados são **atualizados diariamente** deste janeiro de 2020 com uma **granularidade temporal de dias e geográfica de países**. O website do projeto pode ser acessado neste [link](https://ourworldindata.org) enquanto os dados, neste [link](https://covid.ourworldindata.org/data/owid-covid-data.csv). Abaixo estão descritos os dados derivados do seu processamento.

- **date**: data de referência;
 - **country**: país;
 - **population**: população estimada;
 - **total**: número acumulado de doses administradas;
 - **one_shot**: número acumulado de pessoas com uma dose;
 - **one_shot_perc**: número acumulado relativo de pessoas com uma dose;
 - **two_shots**: número acumulado de pessoas com duas doses;
 - **two_shot_perc**: número acumulado relativo de pessoas com duas doses;
 - **three_shots**: número acumulado de pessoas com três doses;
 - **three_shot_perc**: número acumulado relativo de pessoas com três doses;
 - **month**: mês de referência;
 - **year**: ano de referência.

## 2\. Pacotes e bibliotecas

In [ ]:
import math
from typing import Iterator
from datetime import datetime, timedelta

import numpy as np
import pandas as pd

## 3\. Extração

# 3.1 Casos

In [ ]:
cases = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/01-12-2021.csv', sep=',')

In [ ]:
cases.head()

FIPS Admin2 Province_State Country_Region          Last_Update       Lat  \
0   NaN    NaN            NaN    Afghanistan  2021-01-13 05:22:15  33.93911   
1   NaN    NaN            NaN        Albania  2021-01-13 05:22:15  41.15330   
2   NaN    NaN            NaN        Algeria  2021-01-13 05:22:15  28.03390   
3   NaN    NaN            NaN        Andorra  2021-01-13 05:22:15  42.50630   
4   NaN    NaN            NaN         Angola  2021-01-13 05:22:15 -11.20270   

       Long_  Confirmed  Deaths  Recovered  Active Combined_Key  \
0  67.709953      53584    2301      44608    6675  Afghanistan   
1  20.168300      64627    1252      38421   24954      Albania   
2   1.659600     102641    2816      69608   30217      Algeria   
3   1.521800       8682      86       7930     666      Andorra   
4  17.873900      18343     422      15512    2409       Angola   

   Incident_Rate  Case_Fatality_Ratio  
0     137.647787             4.294192  
1    2245.708527             1.937271  
2     234.067409             2.743543  
3   11236.653077             0.990555  
4      55.811022             2.300605

In [ ]:
#Definido um intervalado de tempo para extrair os dados
def date_range(start_date: datetime, end_date: datetime) -> Iterator[datetime]:
  date_range_days: int = (end_date - start_date).days
  for lag in range(date_range_days):
    yield start_date + timedelta(lag)

In [ ]:
start_date = datetime(2021,  1,  1)
end_date   = datetime(2021, 12, 31)

In [ ]:
#Extraindo as colunas de interesse e as linhas referentes ao Brasil
import pandas as pd
import requests
from datetime import datetime, timedelta

cases = None
cases_is_empty = True

for date in date_range(start_date=start_date, end_date=end_date):

  date_str = date.strftime('%m-%d-%Y')
  data_source_url = f'https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/{date_str}.csv'

  case = pd.read_csv(data_source_url, sep=',')

  case = case.drop(['FIPS', 'Admin2', 'Last_Update', 'Lat', 'Long_', 'Recovered', 'Active', 'Combined_Key', 'Case_Fatality_Ratio'], axis=1)
  case = case.query('Country_Region == "Brazil"').reset_index(drop=True)
  case['Date'] = pd.to_datetime(date.strftime('%Y-%m-%d'))

  if cases_is_empty:
    cases = case
    cases_is_empty = False
  else:
    cases = pd.concat([cases, case], ignore_index=True)

In [ ]:
cases.query('Province_State == "Rio de Janeiro"').head()

Province_State Country_Region  Confirmed  Deaths  Incident_Rate  \
20   Rio de Janeiro         Brazil     435390   25600    2521.815450   
47   Rio de Janeiro         Brazil     435604   25608    2523.054956   
74   Rio de Janeiro         Brazil     435739   25616    2523.836887   
101  Rio de Janeiro         Brazil     439345   25617    2544.723142   
128  Rio de Janeiro         Brazil     443607   25837    2569.409004   

          Date  
20  2021-01-01  
47  2021-01-02  
74  2021-01-03  
101 2021-01-04  
128 2021-01-05

# 3.2 Vacinação

In [ ]:
vaccines = pd.read_csv('https://covid.ourworldindata.org/data/owid-covid-data.csv', sep=',', parse_dates=[3], infer_datetime_format=True)

<ipython-input-48-608e6d2dd56d>:1: FutureWarning: The argument 'infer_datetime_format' is deprecated and will be removed in a future version. A strict version of it is now the default, see https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. You can safely remove this argument.
  vaccines = pd.read_csv('https://covid.ourworldindata.org/data/owid-covid-data.csv', sep=',', parse_dates=[3], infer_datetime_format=True)
<ipython-input-48-608e6d2dd56d>:1: DtypeWarning: Columns (33) have mixed types. Specify dtype option on import or set low_memory=False.
  vaccines = pd.read_csv('https://covid.ourworldindata.org/data/owid-covid-data.csv', sep=',', parse_dates=[3], infer_datetime_format=True)


In [ ]:
vaccines.head()

iso_code continent     location       date  total_cases  new_cases  \
0      AFG      Asia  Afghanistan 2020-01-05          0.0        0.0   
1      AFG      Asia  Afghanistan 2020-01-06          0.0        0.0   
2      AFG      Asia  Afghanistan 2020-01-07          0.0        0.0   
3      AFG      Asia  Afghanistan 2020-01-08          0.0        0.0   
4      AFG      Asia  Afghanistan 2020-01-09          0.0        0.0   

   new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  ...  \
0                 NaN           0.0         0.0                  NaN  ...   
1                 NaN           0.0         0.0                  NaN  ...   
2                 NaN           0.0         0.0                  NaN  ...   
3                 NaN           0.0         0.0                  NaN  ...   
4                 NaN           0.0         0.0                  NaN  ...   

   male_smokers  handwashing_facilities  hospital_beds_per_thousand  \
0           NaN                  37.746                         0.5   
1           NaN                  37.746                         0.5   
2           NaN                  37.746                         0.5   
3           NaN                  37.746                         0.5   
4           NaN                  37.746                         0.5   

   life_expectancy  human_development_index  population  \
0            64.83                    0.511    41128772   
1            64.83                    0.511    41128772   
2            64.83                    0.511    41128772   
3            64.83                    0.511    41128772   
4            64.83                    0.511    41128772   

   excess_mortality_cumulative_absolute  excess_mortality_cumulative  \
0                                   NaN                          NaN   
1                                   NaN                          NaN   
2                                   NaN                          NaN   
3                                   NaN                          NaN   
4                                   NaN                          NaN   

   excess_mortality  excess_mortality_cumulative_per_million  
0               NaN                                      NaN  
1               NaN                                      NaN  
2               NaN                                      NaN  
3               NaN                                      NaN  
4               NaN                                      NaN  

[5 rows x 67 columns]

In [ ]:
#Selecionando as colunas de interesse e as linhas referentes ao Brasil
vaccines = vaccines.query('location == "Brazil"').reset_index(drop=True)
vaccines = vaccines[['location', 'population', 'total_vaccinations', 'people_vaccinated', 'people_fully_vaccinated', 'total_boosters', 'date']].reset_index(drop = True)

In [ ]:
vaccines.head()

location  population  total_vaccinations  people_vaccinated  \
0   Brazil   215313504                 NaN                NaN   
1   Brazil   215313504                 NaN                NaN   
2   Brazil   215313504                 NaN                NaN   
3   Brazil   215313504                 NaN                NaN   
4   Brazil   215313504                 NaN                NaN   

   people_fully_vaccinated  total_boosters       date  
0                      NaN             NaN 2020-01-05  
1                      NaN             NaN 2020-01-06  
2                      NaN             NaN 2020-01-07  
3                      NaN             NaN 2020-01-08  
4                      NaN             NaN 2020-01-09

## 4\. Transformação

# 4.1 Casos

In [ ]:
cases.head()

Province_State Country_Region  Confirmed  Deaths  Incident_Rate       Date
0           Acre         Brazil      41689     796    4726.992352 2021-01-01
1        Alagoas         Brazil     105091    2496    3148.928928 2021-01-01
2          Amapa         Brazil      68361     926    8083.066602 2021-01-01
3       Amazonas         Brazil     201574    5295    4863.536793 2021-01-01
4          Bahia         Brazil     494684    9159    3326.039611 2021-01-01

In [ ]:
cases.shape

(9828, 6)

In [ ]:
cases.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9828 entries, 0 to 9827
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Province_State  9828 non-null   object        
 1   Country_Region  9828 non-null   object        
 2   Confirmed       9828 non-null   int64         
 3   Deaths          9828 non-null   int64         
 4   Incident_Rate   9828 non-null   float64       
 5   Date            9828 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(2), object(2)
memory usage: 460.8+ KB


In [ ]:
#Renomeando os nomes das colunas
cases = cases.rename(
  columns={
    'Province_State': 'state',
    'Country_Region': 'country'
  }
)

for col in cases.columns:
  cases = cases.rename(columns={col: col.lower()})

In [ ]:
#Ajustando o nome dos Estados
states_map = {
    'Amapa': 'Amapá',
    'Ceara': 'Ceará',
    'Espirito Santo': 'Espírito Santo',
    'Goias': 'Goiás',
    'Para': 'Pará',
    'Paraiba': 'Paraíba',
    'Parana': 'Paraná',
    'Piaui': 'Piauí',
    'Rondonia': 'Rondônia',
    'Sao Paulo': 'São Paulo'
}

cases['state'] = cases['state'].apply(lambda state: states_map.get(state) if state in states_map.keys() else state)

In [ ]:
#Chaves temporais
cases['month'] = cases['date'].apply(lambda date: date.strftime('%Y-%m'))
cases['year']  = cases['date'].apply(lambda date: date.strftime('%Y'))

In [ ]:
#População estimada do Estado
cases['population'] = round(100000 * (cases['confirmed'] / cases['incident_rate']))
cases = cases.drop('incident_rate', axis=1)

In [ ]:
#Calculando número, media móvel(7 dias) e estabilidade (14 dias) de casos e mortes por Estado
cases_ = None
cases_is_empty = True

def get_trend(rate: float) -> str:

  if np.isnan(rate):
    return np.NaN

  if rate < 0.75:
    status = 'downward'
  elif rate > 1.15:
    status = 'upward'
  else:
    status = 'stable'

  return status


for state in cases['state'].drop_duplicates():

  cases_per_state = cases.query(f'state == "{state}"').reset_index(drop=True)
  cases_per_state = cases_per_state.sort_values(by=['date'])

  cases_per_state['confirmed_1d'] = cases_per_state['confirmed'].diff(periods=1)
  cases_per_state['confirmed_moving_avg_7d'] = np.ceil(cases_per_state['confirmed_1d'].rolling(window=7).mean())
  cases_per_state['confirmed_moving_avg_7d_rate_14d'] = cases_per_state['confirmed_moving_avg_7d']/cases_per_state['confirmed_moving_avg_7d'].shift(periods=14)
  cases_per_state['confirmed_trend'] = cases_per_state['confirmed_moving_avg_7d_rate_14d'].apply(get_trend)

  cases_per_state['deaths_1d'] = cases_per_state['deaths'].diff(periods=1)
  cases_per_state['deaths_moving_avg_7d'] = np.ceil(cases_per_state['deaths_1d'].rolling(window=7).mean())
  cases_per_state['deaths_moving_avg_7d_rate_14d'] = cases_per_state['deaths_moving_avg_7d']/cases_per_state['deaths_moving_avg_7d'].shift(periods=14)
  cases_per_state['deaths_trend'] = cases_per_state['deaths_moving_avg_7d_rate_14d'].apply(get_trend)

  if cases_is_empty:
    cases_ = cases_per_state
    cases_is_empty = False
  else:
    cases_ = pd.concat([cases_, cases_per_state], ignore_index=True)

cases = cases_
cases_ = None

In [ ]:
#Fazendo o type casting das colunas
cases['population'] = cases['population'].astype('Int64')
cases['confirmed_1d'] = cases['confirmed_1d'].astype('Int64')
cases['confirmed_moving_avg_7d'] = cases['confirmed_moving_avg_7d'].astype('Int64')
cases['deaths_1d'] = cases['deaths_1d'].astype('Int64')
cases['deaths_moving_avg_7d'] = cases['deaths_moving_avg_7d'].astype('Int64')

In [ ]:
#Reorganizando as colunas
cases = cases[['date', 'country', 'state', 'population', 'confirmed', 'confirmed_1d', 'confirmed_moving_avg_7d', 'confirmed_moving_avg_7d_rate_14d', 'confirmed_trend', 'deaths', 'deaths_1d', 'deaths_moving_avg_7d', 'deaths_moving_avg_7d_rate_14d', 'deaths_trend', 'month', 'year']]

In [ ]:
cases.head(n=25)

/usr/local/lib/python3.10/dist-packages/pandas/core/nanops.py:1010: RuntimeWarning: invalid value encountered in subtract
  sqr = _ensure_numeric((avg - values) ** 2)
/usr/local/lib/python3.10/dist-packages/pandas/core/nanops.py:1010: RuntimeWarning: invalid value encountered in subtract
  sqr = _ensure_numeric((avg - values) ** 2)


date country state  population  confirmed  confirmed_1d  \
0  2021-01-01  Brazil  Acre      881935      41689          <NA>   
1  2021-01-02  Brazil  Acre      881935      41941           252   
2  2021-01-03  Brazil  Acre      881935      42046           105   
3  2021-01-04  Brazil  Acre      881935      42117            71   
4  2021-01-05  Brazil  Acre      881935      42170            53   
5  2021-01-06  Brazil  Acre      881935      42378           208   
6  2021-01-07  Brazil  Acre      881935      42478           100   
7  2021-01-08  Brazil  Acre      881935      42814           336   
8  2021-01-09  Brazil  Acre      881935      42908            94   
9  2021-01-10  Brazil  Acre      881935      43127           219   
10 2021-01-11  Brazil  Acre      881935      43346           219   
11 2021-01-12  Brazil  Acre      881935      43432            86   
12 2021-01-13  Brazil  Acre      881935      43785           353   
13 2021-01-14  Brazil  Acre      881935      44068           283   
14 2021-01-15  Brazil  Acre      881935      44156            88   
15 2021-01-16  Brazil  Acre      881935      44621           465   
16 2021-01-17  Brazil  Acre      881935      44767           146   
17 2021-01-18  Brazil  Acre      881935      44775             8   
18 2021-01-19  Brazil  Acre      881935      45208           433   
19 2021-01-20  Brazil  Acre      881935      45429           221   
20 2021-01-21  Brazil  Acre      881935      45729           300   
21 2021-01-22  Brazil  Acre      881935      45987           258   
22 2021-01-23  Brazil  Acre      881935      46239           252   
23 2021-01-24  Brazil  Acre      881935      46429           190   
24 2021-01-25  Brazil  Acre      881935      46539           110   

    confirmed_moving_avg_7d  confirmed_moving_avg_7d_rate_14d confirmed_trend  \
0                      <NA>                               NaN             NaN   
1                      <NA>                               NaN             NaN   
2                      <NA>                               NaN             NaN   
3                      <NA>                               NaN             NaN   
4                      <NA>                               NaN             NaN   
5                      <NA>                               NaN             NaN   
6                      <NA>                               NaN             NaN   
7                       161                               NaN             NaN   
8                       139                               NaN             NaN   
9                       155                               NaN             NaN   
10                      176                               NaN             NaN   
11                      181                               NaN             NaN   
12                      201                               NaN             NaN   
13                      228                               NaN             NaN   
14                      192                               NaN             NaN   
15                      245                               NaN             NaN   
16                      235                               NaN             NaN   
17                      205                               NaN             NaN   
18                      254                               NaN             NaN   
19                      235                               NaN             NaN   
20                      238                               NaN             NaN   
21                      262                          1.627329          upward   
22                      232                          1.669065          upward   
23                      238                          1.535484          upward   
24                      252                          1.431818          upward   

    deaths  deaths_1d  deaths_moving_avg_7d  deaths_moving_avg_7d_rate_14d  \
0      796       <NA>                  <NA>            

# 4.2 Vacinação

In [ ]:
vaccines.head()

location  population  total_vaccinations  people_vaccinated  \
0   Brazil   215313504                 NaN                NaN   
1   Brazil   215313504                 NaN                NaN   
2   Brazil   215313504                 NaN                NaN   
3   Brazil   215313504                 NaN                NaN   
4   Brazil   215313504                 NaN                NaN   

   people_fully_vaccinated  total_boosters       date  
0                      NaN             NaN 2020-01-05  
1                      NaN             NaN 2020-01-06  
2                      NaN             NaN 2020-01-07  
3                      NaN             NaN 2020-01-08  
4                      NaN             NaN 2020-01-09

In [ ]:
vaccines.shape

(1667, 7)

In [ ]:
vaccines.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1667 entries, 0 to 1666
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   location                 1667 non-null   object        
 1   population               1667 non-null   int64         
 2   total_vaccinations       695 non-null    float64       
 3   people_vaccinated        691 non-null    float64       
 4   people_fully_vaccinated  675 non-null    float64       
 5   total_boosters           455 non-null    float64       
 6   date                     1667 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 91.3+ KB


In [ ]:
#Tratando os dados faltantes
vaccines = vaccines.fillna(method='ffill')

<ipython-input-55-34ba31a52744>:2: FutureWarning: DataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  vaccines = vaccines.fillna(method='ffill')


In [ ]:
#Filtrando a bases de dados de acordo com a coluna date
vaccines = vaccines[(vaccines['date']>= '2021-01-01') & (vaccines['date'] <= '2021-12-31')].reset_index(drop=True)

In [ ]:
#Renomeando as colunas
vaccines = vaccines.rename(
  columns={
    'location': 'country',
    'total_vaccinations': 'total',
    'people_vaccinated': 'one_shot',
    'people_fully_vaccinated': 'two_shots',
    'total_boosters': 'three_shots',
  }
)

In [ ]:
#Chaves temporais
vaccines['month'] = vaccines['date'].apply(lambda date: date.strftime('%Y-%m'))
vaccines['year']  = vaccines['date'].apply(lambda date: date.strftime('%Y'))

In [ ]:
#Dados relativos
vaccines['one_shot_perc'] = round(vaccines['one_shot'] / vaccines['population'], 4)
vaccines['two_shots_perc'] = round(vaccines['two_shots'] / vaccines['population'], 4)
vaccines['three_shots_perc'] = round(vaccines['three_shots'] / vaccines['population'], 4)

In [ ]:
#Fazendo o type casting das colunas
vaccines['population'] = vaccines['population'].astype('Int64')
vaccines['total'] = vaccines['total'].astype('Int64')
vaccines['one_shot'] = vaccines['one_shot'].astype('Int64')
vaccines['two_shots'] = vaccines['two_shots'].astype('Int64')
vaccines['three_shots'] = vaccines['three_shots'].astype('Int64')

In [ ]:
#Reorganizando as colunas
vaccines = vaccines[['date', 'country', 'population', 'total', 'one_shot', 'one_shot_perc', 'two_shots', 'two_shots_perc', 'three_shots', 'three_shots_perc', 'month', 'year']]

In [ ]:
vaccines.tail()

date country  population      total   one_shot  one_shot_perc  \
360 2021-12-27  Brazil   215313504  329011365  165952037         0.7707   
361 2021-12-28  Brazil   215313504  329861730  166062249         0.7713   
362 2021-12-29  Brazil   215313504  330718457  166143380         0.7716   
363 2021-12-30  Brazil   215313504  331164041  166185628         0.7718   
364 2021-12-31  Brazil   215313504  331273910  166195505         0.7719   

     two_shots  two_shots_perc  three_shots  three_shots_perc    month  year  
360  142764283          0.6631     25218893            0.1171  2021-12  2021  
361  142965728          0.6640     25758909            0.1196  2021-12  2021  
362  143282084          0.6655     26219623            0.1218  2021-12  2021  
363  143398692          0.6660     26507937            0.1231  2021-12  2021  
364  143436012          0.6662     26571077            0.1234  2021-12  2021

## 5\. Carregamento

In [ ]:
cases.to_csv('cases.csv', index=False)
vaccines.to_csv('vaccines.csv', index=False)

# 6 Storytelling

Nos primeiros meses de 2021, o Brasil ainda estava lutando para controlar a propagação do coronavírus. O número de casos e mortes continuava a subir de forma alarmante. Janeiro e fevereiro foram particularmente severos, com o país enfrentando a segunda onda da pandemia. A variante P.1, identificada inicialmente em Manaus, mostrou-se altamente contagiosa e contribuiu para um aumento significativo nos casos e mortes.

Até o final de 2021, mais de 75% da população adulta brasileira havia recebido pelo menos uma dose da vacina, e cerca de 66% estava totalmente imunizada com duas doses. Esse avanço na vacinação foi crucial para a redução dos casos graves e mortes, oferecendo uma dose de esperança e sinalizando um possível controle da pandemia.

Os números de vacinação começaram a mostrar seu impacto positivo no segundo semestre de 2021. À medida que a cobertura vacinal aumentava, o número de casos graves e mortes começou a diminuir.